### PR TITLE
fix: track FusedAdam bias correction per parameter

### PR DESCRIPTION
**Pull request by MrCapricornLiu**
FusedAdam increments only the last active parameter's step counter and uses it for every tensor of a given dtype. When a parameter has no gradient on some iterations, the counter used for the other parameters can switch, producing incorrect bias correction. A group with multiple dtypes also advances that same counter once per dtype.

Increment each active parameter's counter once and batch tensors by dtype and step. Parameters that advance together still share a fused kernel call; parameters with different update histories get the appropriate bias correction. The optimizer state format is unchanged.

The added tests cover Adam/AdamW, uniform and mixed FP32/FP16/BF16 groups, intermittent gradients, and a state-dict round trip. A small training regression also compares against PyTorch AdamW when a layer bias is intermittently frozen. On H800, the original code fails all nine new cases; the fix passes those and the six existing reference cases.

```bash
python -m pytest tests/unit/ops/adam/test_adamw.py -k 'fused_adam_matches_reference or fused_adam_parameter_steps or fused_adam_intermittent_training' -v
```

An additional two-GPU FP32 check compares three accumulated updates against a full-model AdamW reference for ZeRO stages 0 and 3, including two ZeRO-3 subgroup sizes. All configurations pass, including saving a full DeepSpeed checkpoint after the first update, reconstructing the engine, restoring model/optimizer state and continuing. Changed-file pre-commit checks pass.

This does not reconstruct counters already written incorrectly by older versions. CPU/NVMe offload, multi-node training, checkpoint recovery with a changed world size, and performance were not tested.

### CODE REVIEW

File: deepspeed/ops/adam/fused_adam.py [hunk 1]
```diff
@@ -127,15 +127,11 @@ def step(self, closure=None, grads=None, output_params=None, scale=None, grad_no
             bias_correction = 1 if group['bias_correction'] else 0
             beta1, beta2 = group['betas']
 
-            # assume same step across group now to simplify things
-            # per parameter step can be easily support by making it tensor, or pass list into kernel
+            # Group-level counters are retained for loading older checkpoints.
```
[the remark this answers is not in the document]
Comment (MrCapricornLiu) [reply deepspeed/ops/adam/fused_adam.py:130]: Commit 01832c837640290e7bc720e27b07e3062927d772 already includes `Signed-off-by: Chenghao Liu <chliu@stu.pku.edu.cn>`, and the DCO check is passing. No commit change is needed for this finding.
